pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -140,237 +180,133 @@ func (Plugin) OnInboundRouteConfiguration(in *plugin.InputParams, routeConfigura
-	mesh := env.Mesh
-	if mesh.MixerCheckServer == "" && mesh.MixerReportServer == "" {
-		return nil
-	}
-
-	c := buildTCPMixerFilterConfig(mesh, *node, instance)
-	return &listener.Filter{
-		Name:   v1.MixerFilter,
-		Config: util.MessageToStruct(c),
-	}
-}
-
-// defined in install/kubernetes/helm/istio/charts/mixer/templates/service.yaml
-const (
-	//mixerPortName       = "grpc-mixer"
-	mixerPortNumber = 9091
-	//mixerMTLSPortName   = "grpc-mixer-mtls"
-	mixerMTLSPortNumber = 15004
-)
-
-// buildHTTPMixerFilterConfig builds a mixer HTTP filter config. Mixer filter uses outbound configuration by default
-// (forward attributes, but not invoke check calls)  ServiceInstances belong to the Node.
-func buildHTTPMixerFilterConfig(mesh *meshconfig.MeshConfig, role model.Proxy, nodeInstances []*model.ServiceInstance, outboundRoute bool, config model.IstioConfigStore) *mccpb.HttpClientConfig { // nolint: lll
-	mcs, _, _ := net.SplitHostPort(mesh.MixerCheckServer)
-	mrs, _, _ := net.SplitHostPort(mesh.MixerReportServer)
+func buildTransport(mesh *meshconfig.MeshConfig, uid attribute) *mccpb.TransportConfig {
+	policy, _, _ := net.SplitHostPort(mesh.MixerCheckServer)
+	telemetry, _, _ := net.SplitHostPort(mesh.MixerReportServer)
 
 	port := mixerPortNumber
 	if mesh.AuthPolicy == meshconfig.MeshConfig_MUTUAL_TLS {
 		port = mixerMTLSPortNumber
 	}
 
-	// TODO: derive these port types.
-	transport := &mccpb.TransportConfig{
-		CheckCluster:  model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(mcs), port),
-		ReportCluster: model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(mrs), port),
+	return &mccpb.TransportConfig{
+		CheckCluster:  model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(policy), port),
+		ReportCluster: model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(telemetry), port),
+		// internal telemetry forwarding
+		AttributesForMixerProxy: &mpb.Attributes{Attributes: attributes{
+			"source.uid": uid,
+		}},
 	}
+}
 
-	mxConfig := &mccpb.HttpClientConfig{
-		MixerAttributes: &mpb.Attributes{
-			Attributes: map[string]*mpb.Attributes_AttributeValue{
-				"context.reporter.uid": {
-					Value: &mpb.Attributes_AttributeValue_StringValue{StringValue: "kubernetes://" + role.ID},
+func buildOutboundHTTPFilter(mesh *meshconfig.MeshConfig, attrs attributes, node *model.Proxy) *http_conn.HttpFilter {
+	return &http_conn.HttpFilter{
+		Name: mixer,
+		Config: util.MessageToStruct(&mccpb.HttpClientConfig{
+			MixerAttributes: &mpb.Attributes{Attributes: attrs},
+			ForwardAttributes: &mpb.Attributes{Attributes: attributes{
+				"source.uid": attrUID(node),
+			}},
+			DefaultDestinationService: UnknownDestination,
+			ServiceConfigs: map[string]*mccpb.ServiceConfig{
+				UnknownDestination: {
+					DisableCheckCalls: true, /* TODO: enable client-side checks */
+					MixerAttributes: &mpb.Attributes{Attributes: attributes{ // TODO: fall through destination servicec
```

Comment:
for outbound http, shouldn't we be able to add more destination attributes ?  Or is this meant to happen in the per-route config (where I believe there is a TODO)?

Reply:
Yeap. This is the default list of attributes, for which destination is not known. Since we can embed per-route config (no hook in pilot), we have to leave it unknown here.